server-core/persistence/src/main/java/io/infectnet/server/persistence/impl/InMemoryTokenStorageImpl.java

```diff
@@ -3,28 +3,52 @@
 import io.infectnet.server.persistence.Token;
 import io.infectnet.server.persistence.TokenStorage;
 
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
+/**
+ * {@link TokenStorage} in-memory implementation.
+ */
 public class InMemoryTokenStorageImpl implements TokenStorage {
+
+
+    private List<Token> tokens;
+
+    public InMemoryTokenStorageImpl(List<Token> tokenList) {
```

Comment:
Is the `tokenList` parameter really needed? In my opinion, you should omit the parameter and initialize the `tokens` field with a new list.